napari/plugins/tests/fixtures/napari_test_plugin.py

```diff
@@ -0,0 +1,16 @@
+"""
+Example plugin for testing plugin discovery and loading
+"""
+import pluggy
+
+hookimpl = pluggy.HookimplMarker("napari")
+
+
+def reader_function():
+    raise IOError('it worked')
+
+
+@hookimpl
+def napari_get_reader(path):
+    if path.endswith('true'):
+        return reader_function
```

Comment:
It feels a bit weird to have the plugin raising an IOError when it's supposed (?) to be the working plugin? Could you maybe document this file a bit more in terms of the purpose of it?

I'm also confused about why this is in "fixtures"...?

Reply:
sorry, that was an incomplete "exception bomb" idea that I was going to finish in #937.  Will remove for now.

These are in fixtures because they are "example plugins" that allow us to test various things about plugin discovery.  They are not themselves tests...  they are just a set of fixed/static objects. (and I guess that's how I've seen that handled elsewhere).

Reply:
But are they fixtures in the pytest sense that keyword arguments with the same name as the fixture automatically resolve to the fixture? That's what's confusing me...

Reply:
ah, sorry no.  They are not decorated as `@pytest.fixtures`.  The way they are used/discovered is in these line (the plugin manager is a fixture... not the actual plugin modules):
https://github.com/napari/napari/blob/d1f741d21e1b1400045b40d4d72f58b25ee671ad/napari/plugins/tests/test_manager.py#L10-L13